main.tf

```diff
@@ -15,56 +15,51 @@
 resource "google_compute_route" "route" {
-  count      = "${var.cr_name == "" ? var.tunnel_count  * length(var.remote_subnet):0}"
-  name       = "${google_compute_vpn_gateway.vpn_gateway.name}-tunnel${(count.index%var.tunnel_count)+1}-route${count.index%length(var.remote_subnet)+1}"
-  network    = "${var.network}"
-  project    = "${var.project_id}"
-  dest_range = "${element(var.remote_subnet, (count.index%length(var.remote_subnet)))}"
-  priority   = "${var.route_priority}"
+  count      = var.cr_name == "" ? var.tunnel_count * length(var.remote_subnet) : 0
+  name       = "${google_compute_vpn_gateway.vpn_gateway.name}-tunnel${count.index % var.tunnel_count + 1}-route${count.index % length(var.remote_subnet) + 1}"
+  network    = var.network
+  project    = var.project_id
+  dest_range = element(var.remote_subnet, count.index % length(var.remote_subnet))
+  priority   = var.route_priority
 
-  next_hop_vpn_tunnel = "${google_compute_vpn_tunnel.tunnel-static.*.self_link[count.index%var.tunnel_count]}"
+  next_hop_vpn_tunnel = google_compute_vpn_tunnel.tunnel-static[count.index % var.tunnel_count].self_link
 
-  depends_on = [
-    "google_compute_vpn_tunnel.tunnel-static",
-  ]
+  depends_on = [google_compute_vpn_tunnel.tunnel-static]
 }
 
 # For VPN gateways routing through BGP and Cloud Routers
 ## Create Router Interfaces
 resource "google_compute_router_interface" "router_interface" {
-  count      = "${var.cr_name != "" ? var.tunnel_count : 0}"
+  count      = var.cr_name != "" ? var.tunnel_count : 0
   name       = "interface-${local.tunnel_name_prefix}-${count.index}"
-  router     = "${var.cr_name}"
-  region     = "${var.region}"
-  ip_range   = "${element(var.bgp_cr_session_range, count.index)}"
-  vpn_tunnel = "${google_compute_vpn_tunnel.tunnel-dynamic.*.name[count.index]}"
-  project    = "${var.project_id}"
+  router     = var.cr_name
+  region     = var.region
+  ip_range   = element(var.bgp_cr_session_range, count.index)
+  vpn_tunnel = google_compute_vpn_tunnel.tunnel-dynamic[count.index].name
+  project    = var.project_id
 
-  depends_on = [
-    "google_compute_vpn_tunnel.tunnel-dynamic",
-  ]
+  depends_on = [google_compute_vpn_tunnel.tunnel-dynamic]
 }
 
 ## Create Peers
 resource "google_compute_router_peer" "bgp_peer" {
-  count                     = "${var.cr_name != "" ? var.tunnel_count : 0}"
+  count                     = var.cr_name != "" ? var.tunnel_count : 0
   name                      = "bgp-session-${count.index}"
-  router                    = "${var.cr_name}"
-  region                    = "${var.region}"
-  peer_ip_address           = "${element(var.bgp_remote_session_range, count.index)}"
-  peer_asn                  = "${element(var.peer_asn, count.index)}"
-  advertised_route_priority = "${var.advertised_route_priority}"
+  router                    = var.cr_name
+  region                    = var.region
+  peer_ip_address           = element(var.bgp_remote_session_range, [count.index])
```

Comment:
Why is `count.index` wrapped in a list? This is not valid syntax for `element`.

Reply:
Done